chain/chain/src/resharding.rs

```diff
@@ -39,12 +38,27 @@ pub struct StateSplitResponse {
     pub new_state_roots: Result<HashMap<ShardUId, StateRoot>, Error>,
 }
 
+fn get_trie_update_batch(
+    iter: &mut impl Iterator<Item = (Vec<u8>, Vec<u8>)>,
+) -> Option<Vec<(Vec<u8>, Option<Vec<u8>>)>> {
```

Comment:
1. I think `Option<Vec<(Vec<u8>, Option<Vec<u8>>)>>` is way too cryptic of a type. Could we extract something in here to a named type?
2. I'm not sure why the inner one needs to be Option, especially since this function does not return anything that has a None inside. A comment would help.